OpenRA.Mods.Common/Traits/AutoTarget.cs

```diff
@@ -256,6 +256,10 @@ void INotifyDamage.Damaged(Actor self, AttackInfo e)
 			if (attacker.AppearsFriendlyTo(self))
 				return;
 
+			// When stop order given while stance is greater than return fire, it retargets to the closest unit.
+			if (stance > UnitStance.ReturnFire)
+				attacker = ScanForTarget(self, allowMove, true) != Target.Invalid ? ScanForTarget(self, allowMove, true).Actor : attacker;
```

Comment:
We don't need to scan twice. It's not a cheap operation.

```suggestion
			{
				var autoTarget = ScanForTarget(self, allowMove, true);
				if (autoTarget != Target.Invalid)
					attacker = autoTarget.Actor;
			}
```